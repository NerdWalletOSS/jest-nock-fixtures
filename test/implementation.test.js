const fetch = require('node-fetch');
// import fetch from 'node-fetch';

const { expect } = global;

// this server responds with a current-time-in-ms in each response body
const TEST_URL = 'http://worldclockapi.com/api/json/utc/now';

describe('implementation', () => {
  it('should not allow external requests in CI', async () => {
    // const res = await fetch(TEST_URL);
    let res = await fetch(TEST_URL);
    const json = await res.json();
    expect(json).toMatchSnapshot();
    res = await fetch('https://jsonplaceholder.typicode.com/todos/1');
    await res.json();
  });

  it('Hi here is another', async () => {
    // const res = await fetch('http://worldclockapi.com/api/json/est/now');
    const res = await fetch('https://jsonplaceholder.typicode.com/todos/1');
    // https://jsonplaceholder.typicode.com/todos/1
    const json = await res.json();
    expect(json).toMatchSnapshot();
  });

  it('AND here is another', async () => {
    // const res = await fetch('http://worldclockapi.com/api/json/est/now');
    const res = await fetch('https://jsonplaceholder.typicode.com/todos/1');
    // https://jsonplaceholder.typicode.com/todos/1
    const json = await res.json();
    expect(json).toMatchSnapshot();
  });

  it('AND YET here is another', async () => {
    const res = await fetch('http://worldclockapi.com/api/json/est/now');
    // const res = await fetch('https://jsonplaceholder.typicode.com/todos/1');
    // https://jsonplaceholder.typicode.com/todos/1
    const json = await res.json();
    expect(json).toMatchSnapshot();
  });
  it('AND YET here is another', async () => {
    const res = await fetch('http://worldclockapi.com/api/json/est/now');
    // const res = await fetch('https://jsonplaceholder.typicode.com/todos/1');
    // https://jsonplaceholder.typicode.com/todos/1
    const json = await res.json();
    expect(json).toMatchSnapshot();
  });
  it('has no requests', async () => {
    // const res = await fetch('http://worldclockapi.com/api/json/est/now');
    // // const res = await fetch('https://jsonplaceholder.typicode.com/todos/1');
    // // https://jsonplaceholder.typicode.com/todos/1
    // const json = await res.json();
    // expect(json).toMatchSnapshot();
    expect(2).toBe(2);
  });
});
