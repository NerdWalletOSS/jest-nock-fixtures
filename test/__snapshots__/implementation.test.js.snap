// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`implementation AND YET here is another 1`] = `
Object {
  "$id": "1",
  "currentDateTime": "2021-03-31T14:12-04:00",
  "currentFileTime": 132616735589740180,
  "dayOfTheWeek": "Wednesday",
  "isDayLightSavingsTime": true,
  "ordinalDate": "2021-90",
  "serviceResponse": null,
  "timeZoneName": "Eastern Standard Time",
  "utcOffset": "-04:00:00",
}
`;

exports[`implementation AND YET here is another 2`] = `
Object {
  "$id": "1",
  "currentDateTime": "2021-03-31T14:12-04:00",
  "currentFileTime": 132616735593021360,
  "dayOfTheWeek": "Wednesday",
  "isDayLightSavingsTime": true,
  "ordinalDate": "2021-90",
  "serviceResponse": null,
  "timeZoneName": "Eastern Standard Time",
  "utcOffset": "-04:00:00",
}
`;

exports[`implementation AND here is another 1`] = `
Object {
  "completed": false,
  "id": 1,
  "title": "delectus aut autem",
  "userId": 1,
}
`;

exports[`implementation Hi here is another 1`] = `
Object {
  "completed": false,
  "id": 1,
  "title": "delectus aut autem",
  "userId": 1,
}
`;

exports[`implementation should not allow external requests in CI 1`] = `
Object {
  "$id": "1",
  "currentDateTime": "2021-03-31T18:12Z",
  "currentFileTime": 132616879580677580,
  "dayOfTheWeek": "Wednesday",
  "isDayLightSavingsTime": false,
  "ordinalDate": "2021-90",
  "serviceResponse": null,
  "timeZoneName": "UTC",
  "utcOffset": "00:00:00",
}
`;
